test-validator/src/lib.rs

```diff
@@ -852,23 +855,21 @@ impl TestValidator {
             genesis_config.ticks_per_slot = ticks_per_slot;
         }
 
-        // Remove features tagged to deactivate
-        for deactivate_feature_pk in &config.deactivate_feature_set {
-            if FEATURE_NAMES.contains_key(deactivate_feature_pk) {
-                match genesis_config.accounts.remove(deactivate_feature_pk) {
-                    Some(_) => info!("Feature for {:?} deactivated", deactivate_feature_pk),
-                    None => warn!(
-                        "Feature {:?} set for deactivation not found in genesis_config account list, ignored.",
-                        deactivate_feature_pk
-                    ),
-                }
+        // Only activate features which are not explicitly deactivated.
+        let mut feature_set = FeatureSet::default().inactive;
+        for feature in &config.deactivate_feature_set {
+            if feature_set.remove(feature) {
+                info!("Feature for {:?} deactivated", feature)
             } else {
                 warn!(
                     "Feature {:?} set for deactivation is not a known Feature public key",
-                    deactivate_feature_pk
-                );
+                    feature,
+                )
             }
         }
+        for feature in feature_set {
+            genesis_utils::activate_feature(&mut genesis_config, feature);
+        }
```

Comment:
This may look a little redundant, but I wanted to be absolutely sure the original behavior for `create_genesis_config_with_leader_ex` was preserved.